incubator/mongodb-replicaset/init/files/on-start.sh

```diff
@@ -14,48 +14,85 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-function join {
-    local IFS="$1"; shift; echo "$*";
+replica_set=$REPLICA_SET
+script_name=${0##*/}
+
+if [[ "$AUTH" == "true" ]]; then
+    admin_user="$ADMIN_USER"
+    admin_password="$ADMIN_PASSWORD"
+    admin_auth=(-u "$admin_user" -p "$admin_password")
+fi
+
+function log() {
+    local msg="$1"
+    local timestamp=$(date --iso-8601=ns)
+    echo "[$timestamp] [$script_name] $msg" >> /work-dir/log.txt
+}
+
+function shutdown_mongo() {
+    log "Shutting down MongoDB..."
+    mongo admin "${admin_auth[@]}" --eval 'db.shutdownServer({force: true})'
```

Comment:
@foxish Actually, I had experimented with timeouts but gave up on them because they did not work in all cases. That would probably need a more elaborate solution. I can try and look into that but would prefer to do it in a separate PR.

From the MongoDB docs:

> If the node you’re trying to shut down is a replica set primary, then the command will succeed only if there exists a secondary node whose oplog data is within 10 seconds of the primary. You can override this protection using the force option.